src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -331,7 +330,6 @@ void GcsPlacementGroupManager::RemovePlacementGroup(
   }
   auto placement_group = std::move(placement_group_it->second);
   registered_placement_groups_.erase(placement_group_it);
-  placement_group_to_create_callbacks_.erase(placement_group_id);
 
```

Comment:
sorry, I didn't get that, I think we are making the Put operation sync in this pr?

Reply:
`gcs_table_storage_->PlacementGroupTable().Put(` is async. So it is possible remove request comes before registration is recorded to Redis. In that case, we shouldn't return to users that the creation was successful.

Reply:
Imagine this; 

1. Registration comes
2. Async put is called. placement_group_to_register_callbacks_ has the callback now. We haven't replied to the user yet.
3. Remove request comes. 
4. After async put is done, since we didn't remove a callback from `placement_group_to_register_callbacks_`, we will reply to the caller that the placement group registration succeeds, which is wrong.